hybris/tests/test_camera.c

```diff
@@ -286,6 +292,76 @@ static GLuint create_program(const char* pVertexSource, const char* pFragmentSou
 	return program;
 }
 
+
+struct wl_display *wldisplay = NULL;
+struct wl_compositor *wlcompositor = NULL;
+struct wl_surface *wlsurface;
+struct wl_egl_window *wlegl_window;
+struct wl_region *wlregion;
+struct wl_shell *wlshell;
+struct wl_shell_surface *wlshell_surface;
+
+static void
+global_registry_handler(void *data, struct wl_registry *registry, uint32_t id,
+	       const char *interface, uint32_t version)
+{
+    printf("Got a registry event for %s id %d\n", interface, id);
+    if (strcmp(interface, "wl_compositor") == 0) {
+        wlcompositor = wl_registry_bind(registry, 
+				      id, 
+				      &wl_compositor_interface, 
+				      1);
+    } else if (strcmp(interface, "wl_shell") == 0) {
+	wlshell = wl_registry_bind(registry, id,
+				 &wl_shell_interface, 1);
+	
+    }
+}
+
+static void
+global_registry_remover(void *data, struct wl_registry *registry, uint32_t id)
+{
+    printf("Got a registry losing event for %d\n", id);
+}
+
+static const struct wl_registry_listener registry_listener = {
+    global_registry_handler,
+    global_registry_remover
+};
+
+static void
+get_server_references(void) {
+
+    wldisplay = wl_display_connect(NULL);
+    if (wldisplay == NULL) {
+	fprintf(stderr, "Can't connect to display\n");
+	exit(1);
+    }
+    printf("connected to display\n");
+
+    struct wl_registry *registry = wl_display_get_registry(wldisplay);
+    wl_registry_add_listener(registry, &registry_listener, NULL);
+
+    wl_display_dispatch(wldisplay);
+    wl_display_roundtrip(wldisplay);
+
+    if (wlcompositor == NULL || wlshell == NULL) {
+	fprintf(stderr, "Can't find compositor or shell\n");
+	exit(1);
+    } else {
+	fprintf(stderr, "Found compositor and shell\n");
+    }
+}
+
+static void
+create_opaque_region() {
```

Comment:
Please use consistent function formatting, all old parts had the return value type and function name on same line and the starting curly bracket on the next line. Same goes for other functions you added.